src/ConfigGen/Program.cs

```diff
@@ -322,6 +322,33 @@ static string EnumNameToDotnetName(string enumName)
             return result;
         }
 
+        static string createConfigKeys(IEnumerable<PropertySpecification> props)

```

Comment:
Thanks for these additions!

There's an existing class `ConfigPropertyNames` (which predates `Config`) with some dotnet specific names in it. We shouldn't have this and what you've done. It's better under `Config` as you are doing it, but I prefer `PropertyNames` over `KeyNames`. I think we should depreciate `ConfigPropertyNames` (and move to `PropertyNames` under `Config`).

But a problem with your additions is they lets you to do, e.g. `ProducerConfig.GroupId`, which isn't correct. The nested static classes in `ConfigPropertyNames` solves this.

Reply:
I've reworked the codegen. Could you confirm that I'm using `CPorA` the right way ?

Reply:
If we want to "deprecate" `ConfigPropertyNames`, we'd need to also rework the current uses within this project. There are 19 references. Simple enough to do.
Would you want that done in this PR? Or done in a follow-up?

Reply:
Decided to see what it would look like.
Quite neat. Apart from inside the Consumer/Producer classes where the internal `Config` classes mean that `Config.PropertyNames` class needs to be disambiguated. I think it's ok.
Thoughts?

Reply:
ping?